Text sizescale

danrabbit opened the pull request:
- [x] Based on #297

Fixes #289
Fixes #248 

![Screenshot from 2021-11-10 12 45 08](https://user-images.githubusercontent.com/7277719/141190635-d20f3bec-7025-49db-b11f-5d9e158aaf44.png)

cassidyjames commented on src/Views/Text.vala:
```diff
@@ -70,52 +70,51 @@ public class PantheonShell.Text : Gtk.Grid {
-        text_size_modebutton.mode_changed.connect (() => {
-            interface_settings.set_double (TEXT_SIZE_KEY, TEXT_SCALE[text_size_modebutton.selected]);
+        // Setting scale is slow, so we wait to keep UI responsive
+        text_size_scale.button_release_event.connect (() => {
```
So the problem with this approach is that it's not accessible with the keyboard. Gotta keep that. :smile:

cassidyjames commented on src/Views/Text.vala:
```diff
@@ -30,20 +28,22 @@ public class PantheonShell.Text : Gtk.Grid {
+        text_size_scale.add_mark (1, Gtk.PositionType.BOTTOM, _("Default"));
+        text_size_scale.add_mark (1.25, Gtk.PositionType.BOTTOM, null);
+        // Avoid rendering bug when putting mark at the very end
+        text_size_scale.add_mark (1.45, Gtk.PositionType.BOTTOM, _("Largest"));
```
This makes me wonder if we need the small and largest marks at all; they're kind of self-evident based on the design, and we would avoid this funkiness.

cassidyjames reviewed (changes requested):
Need to fix keyboard interactivity, I think I messed up resolving merge conflicts (and will try to fix that), and I have a question about the design:

Should we just go with the scale + spinbutton like we do in both Keyboard and Mouse & Touchpad settings? It seems like this is an appropriate place and would make it easier if you know a specific text scale works well on your machine versus having to just guess based on the current design. I also think it would let us address my other inline comment about the start and end marks, since the scale would be even more self-evident.

danrabbit commented:
Yeah good call about the spin button

danrabbit commented:
@cassidyjames thanks for the thorough review here! I changed things to match the design of scale + spinbutton we have in other views like Mouse & Touchpad. So, removed the labels, notches on top. I also changed the set method from button release on the scale to a timeout on the adjustment. This fixes the keyboard case and should still have the same effect of keeping the UI from doing a stuttering resize while you're still dragging

cassidyjames reviewed (approved):
Nice work, this seems a lot better.
